allennlp/common/file_utils.py

```diff
@@ -63,6 +65,38 @@
     )
 
 
+class FileLock(_FileLock):
+    """
+    This is just a subclass of the `FileLock` class from the `filelock` library, except that
+    it adds an additional argument to the `__init__` method: `read_only_ok`.
+
+    By default this flag is `False`, which an exception will be thrown when a lock
+    can't be acquired due to lack of write permissions.
+    But if this flag is set to `True`, a warning will be emitted instead of an error when
+    the lock already exists but the lock can't be acquired because write access is blocked.
+    """
+
+    def __init__(
+        self, lock_file: Union[str, PathLike], timeout=-1, read_only_ok: bool = False
+    ) -> None:
+        super().__init__(str(lock_file), timeout=timeout)
+        self._read_only_ok = read_only_ok
+
+    @overrides
+    def acquire(self, timeout=None, poll_interval=0.05):
+        try:
+            super().acquire(timeout=timeout, poll_intervall=poll_interval)
+        except (PermissionError, OSError):
```

Comment:
Is there a way to be more specific? This could capture all sorts of things.

Reply:
Yeup, good point. Just updated.

Reply:
Oops, `PermissionError` is actually a subclass of `OSError`. Just fixed.